lib/simp/cli/commands/environment/new.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+require 'simp/cli/commands/command'
+require 'simp/cli/environment/omni_env_controller'
+
+# Cli command to create a new Extra/Omni environment
+#
+# TODO: As more `simp environment` sub-commands are added, a lot of this code
+#      could probably be abstracted into a common class or mixin
+class Simp::Cli::Commands::Environment::New < Simp::Cli::Commands::Command
+  TYPES=[:puppet, :secondary, :writable]
+
+  # @return [String] description of command
+  def self.description
+    'Create a new SIMP "Extra" (default) or "omni" environment'
+  end
+
+  # Run the command's `--help` strategy
+  def help
+    parse_command_line(['--help'])
+  end
+
+  # Parse command-line options for this simp command
+  # @param args [Array<String>] ARGV-style args array
+  def parse_command_line(args)
+    # TODO: simp cli should read a config file that can override these defaults
+    # these options (preferrable mimicking cmd-line args)
+    options = Simp::Cli::Utils.default_simp_env_config
+    options[:action] = :create
+
+    opt_parser = OptionParser.new do |opts|
+      opts.banner = '== simp environment new [options]'
+      opts.separator <<-HELP_MSG.gsub(%r{^ {8}}, '')
+
+        #{self.class.description}
+
+        Usage:
+
+          simp environment new ENVIRONMENT [OPTIONS]
+
+        By default, this command will:
+
+          * create a new environment (–-skeleton)
+          * raise an error if an environment directory already exists
+
+        It can create a complete SIMP omni-environment with --puppet-env
+
```

Comment:
Are --skeleton, --link, --copy intended to be mutually exclusive operations?  If so, the USAGE should be more clear.